src/components/v2/appDetails/sourceInfo/scaleWorkloads/ScaleWorkloadsModal.component.tsx

```diff
@@ -0,0 +1,427 @@
+        }
+
+        return tabName
+    }
+
+    const renderScaleWorkloadTabs = (): JSX.Element => {
+        return (
+            <ul className="tab-list deployment-tab-list tab-list--borderd mr-20">
+                {scaleWorkloadTabs.map((tab, index) => {
+                    return (
+                        <li
+                            onClick={() => {
+                                if (!scalingInProgress && !fetchingLatestDetails) {
+                                    changeDeploymentTab(index)
+                                }
+                            }}
+                            key={tab}
+                            className="tab-list__tab"
+                        >
+                            <div
+                                className={`tab-list__tab-link ${selectedDeploymentTabIndex == index ? 'active' : ''}`}
+                                style={{
+                                    cursor: scalingInProgress || fetchingLatestDetails ? 'not-allowed' : 'pointer',
+                                }}
+                            >
+                                {getTabName(tab, index)}
+                            </div>
+                        </li>
+                    )
+                })}
+            </ul>
+        )
+    }
+
+    const handleAllScaleObjectsName = (isActiveWorkloadsTab: boolean): void => {
+        const _nameSelectionKey = isActiveWorkloadsTab ? 'scaleDown' : 'restore'
+        const _nameSelection = nameSelection[_nameSelectionKey]
+        const _workloadsList = isActiveWorkloadsTab ? workloadsToScaleDown : workloadsToRestore
+        const _setWorkloadsList = isActiveWorkloadsTab ? setWorkloadsToScaleDown : setWorkloadsToRestore
+
+        for (let [key, value] of _workloadsList) {
+            value.value = !_nameSelection.isChecked ? 'CHECKED' : 'INTERMEDIATE'
+            value.isChecked = !_nameSelection.isChecked ? true : false
+            _workloadsList.set(key, value)
+        }
+
+        _setWorkloadsList(_workloadsList)
+        setNameSelection({
+            ...nameSelection,
+            [_nameSelectionKey]: {
+                isChecked: !_nameSelection.isChecked,
+                value: _nameSelection.isChecked ? 'INTERMEDIATE' : 'CHECKED',
+            },
+        })
+    }
+
+    const handleWorkloadSelection = (workloadKey: string, isActiveWorkloadsTab: boolean): void => {
+        const _workloadsList = isActiveWorkloadsTab ? workloadsToScaleDown : workloadsToRestore
+        const _setWorkloadsList = isActiveWorkloadsTab ? setWorkloadsToScaleDown : setWorkloadsToRestore
+
+        for (let [key, value] of _workloadsList) {
+            if (workloadKey === `${value.kind}/${value.name}`) {
+                value.value = !value.isChecked ? 'CHECKED' : 'INTERMEDIATE'
+                value.isChecked = !value.isChecked
+            }
+            _workloadsList.set(key, value)
```

Comment:
Can we move this in the if block?

Reply:
Done. Removed the for-loop & if check itself. As we already have the key to fetch the particular value from the map.